builtin/providers/azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -783,6 +876,13 @@ func resourceArmVirtualMachineScaleSetOsProfileLWindowsConfigHash(v interface{})
 	return hashcode.String(buf.String())
 }
 
+func resourceArmVirtualMachineScaleSetExtensionHash(v interface{}) int {
```

Comment:
This function should write all the fields for the extension to ensure the diff works properly.